packages/itwinui-react/src/core/utils/hooks/useId.ts

```diff
@@ -18,4 +18,5 @@ export const useId = () => {
 // This is needed to avoid bundlers trying to import non-existing export.
 // Read more: https://github.com/webpack/webpack/issues/14814
 const _React = React;
-const useUniqueValue = _React.useId ?? (() => getRandomValue(10));
+const useUniqueValue =
+  _React.useId ?? (() => React.useMemo(() => getRandomValue(10), []));
```

Comment:
should we backport this to v2? seems like a major oversight, and most of our users are not using react 18 yet.

Reply:
Yes, I feel we should, especially since it's just 1-2 lines. Although #1679 and #1657 are I think v3 only bugs, this backport might fix some bug that we are not aware of.